riscv-page-tables/src/page_table.rs

```diff
@@ -301,6 +337,12 @@ impl<'a, T: PagingMode> PageTablePte<'a, T> {
         //    `GuestStagePageTable` and have their lifetime bound to the root.
         unsafe { PageTable::from_pte(self.pte, self.level.next().unwrap()) }
     }
+
+    /// Makes the entry a valid leaf.
+    fn leaf(self) -> LeafPte<'a, T> {
+        self.pte.leaf();
+        LeafPte::new(self.pte, self.level)
+    }
```

Comment:
similar to the "unused" case, i don't think this is correct. the PFN of this PTE is currently set to the PFN of the table page. when promoting, we want to set the PFN to the base of the huge page that this PTE is supposed to map.

Reply:
Fixed. PTAL